internal/deviations/deviations.go

```diff
@@ -12,22 +12,27 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package deviations defines the arguments to enable temporary workarounds
-// for the featureprofiles test suite.
+// Package deviations defines the arguments to enable temporary workarounds for the
+// featureprofiles test suite.
 //
-// Deviations may be introduced to temporarily work around non-compliant issues
-// so further sub-tests can be implemented.  Deviations should be
-// small in scope, typically affecting one sub-test, one OpenConfig
-// path or small OpenConfig sub-tree.  Deviations are enabled using
-// command line parameters.
+// A test may introduce deviations in order to temporarily work around non-compliant
+// issues so further sub-tests can make progress.  Deviations should be small in scope,
+// typically affecting one sub-test, one OpenConfig path or small OpenConfig sub-tree.
+// Deviations are enabled using command line parameters.
 //
-// Passing with a deviation enabled is considered non-compliant to the
-// OpenConfig featureprofiles test.
+// The intended behavior to be tested should always be without the deviation, so if a
+// device could not pass without deviation, that is considered non-compliant behavior.
+// Ideally, device should pass both with and without a deviation which means the deviation
```

Comment:
I think we should be very clear that passing with deviation is always non-compliant.  The additional information is good description to elaborate further on behavior when deviations are enabled, but we should keep some simple statement that passing with deviation is non-compliant.

Reply:
What you are saying is that a device that passes without deviation is still non-compliant if it happens to also pass with deviation, and this is not correct.

Most of the existing deviations are implemented using only fully compliant behavior. The only exception so far is the one we will have to reintroduce for #383 to omit setting L2 MTU as a deviation. That's because if we have these scenarios:

  - Device A: enforces that L2 MTU >= L3 MTU and L2 MTU default is 1514. (compliant)
  - Device B: allows L3 MTU to override L2 MTU. (also compliant)
  - Device C: can only set L3 MTU but not L2 MTU (not compliant)

Here is the matrix for a deviation that omits the L2 MTU, for a test that wants to set L3 MTU to 9000.

| Device  | Without Deviation | With Deviation |
| ---- | ---- | ---- |
| Device A  | Pass | Fail (L3 MTU 9000 > L2 MTU 1514) |
| Device B | Pass | Pass |
| Device C | Fail (cannot set L2 MTU) | Pass |

As you can see, if we say "passing with deviation is always non-compliant" then you would disqualify Device B.